storage/allocator.go

```diff
@@ -251,9 +251,9 @@ func (a *Allocator) AllocateTarget(required roachpb.Attributes, existing []roach
 			return target, nil
 		}
 		if len(attrs) == 0 {
-			return nil, util.Errorf("unable to allocate a target store; no candidates available")
+			return nil, util.Errorf("no suitable replication target store found, are you running enough nodes?")
```

Comment:
@petermattis I'll tackle this one tomorrow.

But it's not as simple as knowing the number of nodes (which as @mrtracy said, is available via gossip or potentially the store pool). We need to tailor the message differently for systems that are starting up vs those that are experiencing a partition or multiple crashes.